### ISSUE TITLE
Cannot Start Docker for windows key already exists

### ISSUE THREAD

**Issue by interpeix** [docker/for-win#1701]
### Actual behavior
[10:00:22.525][VpnKit         ][Info   ] vpnkit.exe: AF_HVSOCK connected successfully

[10:00:22.525][VpnKit         ][Info   ] vpnkit.exe: Vmnet.Server.negotiate: received { magic = VMN3T; version = 22; commit =                                          }

[10:00:22.536][VpnKit         ][Info   ] vpnkit.exe: Generated UUID on behalf of client: f86c7321-b091-489f-99bb-336f0f5fbf70

[10:00:22.537][VpnKit         ][Info   ] vpnkit.exe: Vmnet.Server.negotiate: received Ethernet f86c7321-b091-489f-99bb-336f0f5fbf70

[10:00:22.537][VpnKit         ][Info   ] vpnkit.exe: Vmnet.Server.negotiate: sending { mtu = 1500; max_packet_size = 1550; client_macaddr = 02:50:00:00:00:01 }

[10:00:22.545][VpnKit         ][Info   ] vpnkit.exe: Vmnet.Server.listen: rebinding the primary listen callback

[10:00:22.545][VpnKit         ][Info   ] vpnkit.exe: Vmnet.Server.listen: starting event loop

[10:00:22.545][VpnKit         ][Info   ] vpnkit.exe: Connected Ethernet interface f6:16:36:bc:f9:c6

[10:00:22.551][VpnKit         ][Info   ] vpnkit.exe: Client mac: 02:50:00:00:00:01 server mac: f6:16:36:bc:f9:c6

[10:00:22.551][VpnKit         ][Info   ] vpnkit.exe: TCP/IP ready

[10:00:22.551][VpnKit         ][Info   ] vpnkit.exe: TCP/IP stack connected

[10:00:22.551][VpnKit         ][Info   ] vpnkit.exe: Starting diagnostics server on: \\.\pipe\dockerVpnKitDiagnostics

[10:00:22.558][VpnKit         ][Info   ] vpnkit.exe: BOOTREQUEST from 26:7b:39:7d:34:a1

[10:00:22.561][VpnKit         ][Info   ] vpnkit.exe: BOOTREPLY to 26:7b:39:7d:34:a1 yiddr 192.168.65.3 siddr 192.168.65.1 dns 192.168.65.1 router 192.168.65.1 domain innovation.local

[10:00:22.659][Moby           ][Info   ] [34mINFO[0m[0006] shim containerd-shim started                  [34maddress[0m="/containerd-shim/services.linuxkit/vsudd/shim.sock" [34mdebug[0m=false [34mmodule[0m="containerd/tasks" [34mpid[0m=2078
[10:00:22.745][Moby           ][Info   ] [   28.991324] Initializing XFRM netlink socket
[10:00:23.005][Moby           ][Info   ] [   29.255696] IPv6: ADDRCONF(NETDEV_UP): docker0: link is not ready
[10:00:23.214][Linux          ][Info   ] Vsudd is reachable!
[10:00:23.223][NamedPipeServer][Error  ] Unable to execute Start: An item with the same key has already been added.    at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at Docker.Backend.Processes.ApiProxy.GetEnvVars() in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Backend\Processes\ApiProxy.cs:line 56
   at Docker.Backend.Processes.ApiProxy.StartWithVm(Settings settings, String vmId) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Backend\Processes\ApiProxy.cs:line 68
   at Docker.Backend.ContainerEngine.Linux.DoStart(Settings settings, String daemonOptions) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Backend\ContainerEngine\Linux.cs:line 252
   at Docker.Backend.ContainerEngine.Linux.Start(Settings settings, String daemonOptions) in C:\gopath\src\github.com\docker\pinata\win\src\Docker.Backend\ContainerEngine\Linux.cs:line 112
[10:00:23.352][NamedPipeServer][Info   ] Version()
[10:00:23.371][NamedPipeServer][Info   ] Version done in 00:00:00.0190021.
[10:00:23.412][NamedPipeServer][Info   ] Version()
[10:00:23.412][NamedPipeServer][Info   ] Version done in 00:00:00.
[10:00:23.415][NamedPipeServer][Info   ] GetDebugInfo()
[10:00:23.437][NamedPipeServer][Info   ] DownloadVpnKitLogs(C:\Users\r.ferrer\AppData\Local\Temp\4ao5wicr.4ba\pcap.tar)

### Information
CDF812E7-8845-45D2-83F8-C84C86B8D98B/2018-02-14_10-02-37
Windows 10 Enterprise Version 1709 Build 16299.192
Docker CE Version 18.02.0-ce-win52 (15732)

**Comment by DaveWelling** [comment]
For me this was caused by my environment variables.  IMHO this is a bug.  Here is the code (in Docker.Backend, Version=18.3.0.16762, Culture=neutral, PublicKeyToken=null  --- Docker.Backend.Processes.ApiProxy.GetEnvVars (line 55):
```
    private IDictionary<string, string> GetEnvVars()
    {
      Dictionary<string, string> dictionary = new Dictionary<string, string>();
      foreach (DictionaryEntry environmentVariable in Environment.GetEnvironmentVariables())
      {
        string key;
        string str;
        if ((key = environmentVariable.Key as string) != null && (str = environmentVariable.Value as string) != null)
          dictionary.Add(key, str);
      }
      using (this._recorder.Impersonate())
      {
        string path1 = Paths.FullPath(Paths.UserProfile);
        dictionary.Add("HOME", path1);
        if (!dictionary.ContainsKey("KUBECONFIG"))
          dictionary.Add("KUBECONFIG", Path.Combine(path1, ".kube", "config"));
      }
      return (IDictionary<string, string>) dictionary;
    }
```
The code is attempting to add the HOME environment variable without checking for prior existence in the dictionary.  I don't think it is uncommon for this variable to be defined for users using linux type utilities.
To fix you have to remove the HOME environment variable.  This is probably going to break other applications.

**Comment by dazinator** [comment]
I just hit this too, attempting to install docker on windows:

![image](https://user-images.githubusercontent.com/3176632/37919588-3aea1598-311c-11e8-9f9b-97e0f1374970.png)

**Comment by bernardpumped** [comment]
Hi,
I'm now also incurring this problem, did the system reboot after removing HOME environment variable work.

please advise
regards

**Comment by vdillon** [comment]
Hi,

I'm also seeing this problem on two Windows 10 systems I just tried to upgrade.

Version: 18.03.0-ce-win59 (16762)
Windows 10 Pro Version: 1709 Build: 16299.334

**Comment by dazinator** [comment]
@bernardpumped - Yes removing the HOME environment variable and then rebooting worked (simple logout / login didnt work).

I am a bit concerned though, as I remember certain other tools required this HOME environment variable to be present.. So by removing it, I have now broken them, in order for docker to function.

**Comment by bernardpumped** [comment]
Hi Darrell,
thanks for feedback much appreciated, I decided to switch back to 17.12.0 as can't afford to risk breaking anything else by remove HOME

**Comment by vdillon** [comment]
I also decided to revert to 17.12.0.  My development machines all have HOME set for Git and I don't want to break it.

**Comment by eriawan** [comment]
I have this too.
For now I'm back to 17.12.0 until it is fixed in later release.
It's definitely a regression, it happened before in releases before 17.x.

**Comment by stephen-turner** [comment]
Fixed in 18.03.1.
